File: src/modules/role/services/role-find-all-abilities.service.ts
import { prisma } from '../../../infra/database/prisma.service';
import { AbilityEntity } from './../domain/entities/ability.entity';

class RoleFindAllAbilitiesService {
  async execute(): Promise<AbilityEntity[]> {
    const findAbilities = await prisma.ability.findMany();

    return findAbilities.map((ability) => new AbilityEntity(ability));
  }
}

export const roleFindAllAbilitiesService = new RoleFindAllAbilitiesService();

File: src/main/java/br/com/guizzo/projectbasic/modules/role/services/RoleFindAllAbilitiesService.java
package br.com.guizzo.projectbasic.modules.role.services;

import br.com.guizzo.projectbasic.modules.role.domain.entities.Ability;
import br.com.guizzo.projectbasic.modules.role.repositories.AbilityRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@Service
public class RoleFindAllAbilitiesService {


    @Autowired
    private AbilityRepository abilityRepository;

    public List<Ability> execute(){

        List<Ability> abilities = new ArrayList<>();

        abilityRepository.findAll().forEach(abilities::add);
        
        return abilities;
    }

}
